ntoskrnl/io/iomgr/deviface.c

```diff
@@ -304,7 +305,189 @@ IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
+                        Size,
+                        &Size);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("ZwQueryKey() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    /* Enumerate subkeys (i.e. the different device objects) */
+    for (i = 0; i < fip->SubKeys; i++)
+    {
+        Status = ZwEnumerateKey(InterfaceKey,
+                                i,
+                                KeyBasicInformation,
+                                NULL,
+                                0,
+                                &Size);
+        if (Status == STATUS_NO_MORE_ENTRIES)
+        {
+            break;
+        }
+        else if (!NT_SUCCESS(Status) && Status != STATUS_BUFFER_TOO_SMALL)
+        {
+            DPRINT1("ZwEnumerateKey() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        bip = ExAllocatePoolWithTag(NonPagedPool,
+                                    Size,
+                                    TAG_IO);
+        if (!bip)
+        {
+            DPRINT1("ExAllocatePoolWithTag() failed\n");
+            Status = STATUS_INSUFFICIENT_RESOURCES;
+            goto cleanup;
+        }
+
+        Status = ZwEnumerateKey(InterfaceKey,
+                                i,
+                                KeyBasicInformation,
+                                bip,
+                                Size,
+                                &Size);
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("ZwEnumerateKey() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        /* Check bip->Name */
+        RtlInitUnicodeString(&bipName, NULL);
+        RtlInitUnicodeString(&InLinkName, NULL);
+
+        bipName.Length = wcslen(bip->Name);
+        bipName.MaximumLength = wcslen(bip->Name);
+        bipName.Buffer = bip->Name;
+
+        AliasKeyName.Length = 0;
+        AliasKeyName.MaximumLength += bipName.Length;
+        AliasKeyName.Buffer = ExAllocatePoolWithTag(NonPagedPool,
+                                                    AliasKeyName.MaximumLength,
+                                                    TAG_IO);
+        if (!AliasKeyName.Buffer)
+        {
+            DPRINT1("ExAllocatePoolWithTag() failed\n");
+            Status = STATUS_INSUFFICIENT_RESOURCES;
+            goto cleanup;
+        }
+
+        Status = RtlAppendUnicodeStringToString(&AliasKeyName, &bipName);
+        if (!NT_SUCCESS(Status))
+        {
+            DPRINT1("RtlAppendUnicodeStringToString() failed with status 0x%08lx\n", Status);
+            goto cleanup;
+        }
+
+        pdest = wcsstr(SymbolicLinkName->Buffer, L"Services\\");
+
+        InLinkName.Length = wcslen(pdest);
+        InLinkName.MaximumLength = wcslen(pdest);
+        InLinkName.Buffer = pdest;
+
+        ret = RtlCompareUnicodeString(&bipName,
+                                      &InLinkName,
+                                      TRUE);
+        if (ret != 0)
```

Comment:
```suggestion
        if (RtlCompareUnicodeString(&bipName,
                                    &InLinkName,
                                    TRUE) != 0)
```
and get rid of this `LONG ret;` at the beginning.

Reply:
Committed.